cdk_framework/EKS/lib/cluster-deployment.ts

```diff
@@ -21,30 +25,49 @@ export function deployClusters(app: cdk.App) : Map<string, ClusterStack> {
     const raw = readFileSync(route)
     const configData = yaml.load(raw)
 
-    const eksClusterMap = new Map<string, ClusterStack>();
+    validateFileSchema(configData)
+
+    const eksClusterMap = new Map<string, FargateStack | EC2Stack>();
     
     const vpcStack = new VPCStack(app, 'EKSVpc', {
       env: {
         region: REGION
       }
     })
 
-    validateClustersConfig(configData)
-    for(const [key, value] of Object.entries(configData['clusters'])){
-      const val = Object(value)
-      const versionKubernetes = eks.KubernetesVersion.of(String(val['version']));
-      const newStack = new ClusterStack(app, key + 'EKSCluster', {
-        launchType: (val['launch_type']),
-        name: key,
-        vpc: vpcStack.vpc,
-        version: versionKubernetes,
-        env: {
-          region: REGION
-        },
-      })
-        
-    
-      eksClusterMap.set(key, newStack)
+    const clusterNameSet = new Set()
+    for(const cluster of configData['clusters']){
+      let stack;
+      const clusterInterface = cluster as ClusterInterface
+      if(clusterNameSet.has(clusterInterface.name)){
+        throw new Error(`Cluster name ${clusterInterface.name} is shared by two different clusters`)
+      }
+      clusterNameSet.add(clusterInterface.name)
+      const versionKubernetes = eks.KubernetesVersion.of(clusterInterface.version);
+      if(clusterInterface.launch_type === 'ec2'){
+        const ec2Cluster = cluster as ec2ClusterInterface
+        validateInterface(ec2Cluster)
```

Comment:
Can these `validateInterface` calls be done once before the `if` statement?

Reply:
The reason I do it after is because I want to potentially cast to `ec2ClusterInterface`, so that the validation will also check the `instance_type`